### PR TITLE
Introduce family filter for documents

### PR DESCRIPTION
**Pull request by winged**
This is intended to improve loading speed so you don't need to build a
large nested query anymore, which should speed up data retreival:
```graphql
        query foo ($root: ID!) {
          allDocuments(rootDocument: $root) {
            edges {
              node {
                id
              }
            }
          }
        }
```
This should get you all documents regardless of nesting level. Fetching the corresponding answers for them should then allow you to get the whole structure in a flat result

### CODE REVIEW

**Comment by winged** [comment]
@anehx can you check whether this allows you to improve document query performance?

**Comment by anehx** [comment]
Would be nice if this PR also included #474

**Comment by winged** [comment]
@anehx last commit implements #474

File: caluma/core/forms.py [hunk 1]
```diff
@@ -23,3 +26,20 @@ class GlobalIDMultipleChoiceField(forms.GlobalIDMultipleChoiceField):
 
     def valid_value(self, value):
         return True
+
+
+class SlugMultipleChoiceField(GlobalIDMultipleChoiceField):
+    """
+    Slug multiple choice field which allows listing multiple slugs.
+
+    Disable cleaning as we dont' know which slugs may be valid down
```
Comment (anehx) [inline caluma/core/forms.py]:
```suggestion
    Disable cleaning as we don't know which slugs may be valid down
```

**Pull request merged**